app.py

```diff
@@ -86,15 +64,16 @@ def create_chain(config: dict):
         top_k=config["settings"]["top_k"],
         top_p=config["settings"]["top_p"],
     )
-    # https://api.python.langchain.com/en/latest/chains/langchain.chains.conversational_retrieval.base.ConversationalRetrievalChain.html
-    chain = ConversationalRetrievalChain.from_llm(
-        chain_type="stuff",
-        llm=model,
-        memory=memory,
-        retriever=vectordb.as_retriever(
-            search_kwargs={"k": int(config["settings"]["num_sources"])}
-        ),
-        return_source_documents=True,
+    chain = (
```

Comment:
Create a runnable rather than using a function. I scouted multiple functions that can create History-aware RAG retrievers and do ChatQ&A, but none of them worked as well as just creating the runnable myself.